src/components/RichTextContainer.ts

```diff
@@ -83,21 +105,38 @@ export default class RichTextContainer extends Component<RichTextContainerProps,
                 window.mx.data.subscribe(commonOptions),
                 window.mx.data.subscribe({ attr: this.props.stringAttribute, ...commonOptions })
             ];
+
+            this.props.mxform.listen("submit", this.onFormSubmit);
         }
     }
 
     private handleSubscriptions() {
         this.setState({
             value: getValue(this.props.stringAttribute, "", this.props.mxObject) as string
-        });
+        }, () => this.update = true);
```

Comment:
Why is `this.update` not part of `this.state`?

Reply:
It's purpose wouldn't be served in the state, so this was the only option. Purpose was to control update behaviour between changes triggered by the on change feature & those made externally (changing the value of the attribute outside the widget).